go/Internal/database/connection.go

```diff
@@ -0,0 +1,43 @@
+package database
+
+import (
+	"os"
+	"strings"
+
+	"github.com/gocql/gocql"
+	"github.com/scylladb/gocqlx/v2"
+)
+
+type DbInterface interface {
+	Connect() (*gocqlx.Session, error)
+}
+
+type Db struct {
+
+}
+
+func New() *Db {
+	return &Db{}
+}
+
+func (db *Db) Connect() (*gocqlx.Session, error) {
+	nodes := os.Getenv("NODES")
```

Comment:
I would probably convert all these parameters that you are getting from the environment and transform them to a struct, so I'll inject the Config to the `connect` function.

Why? Simple. I can validate the Config before trying to connect to the database, if the configuration doesn't have all the values that I need I'll immediately fail, not even calling the connect function. So, it means that your error handling would be directly at your `main.go` file.